### PR TITLE
Update Kotlin to 1.4.30 and add a prefix to DispatchQueue

### PR DESCRIPTION
**Pull request by antoinelamy**
## Description
- Migrate kotlin to `1.4.30`
- Remove deprecated `jcenter` repository
- Migrate Gradle scripts to use Kotlin DSL over Groovy
- Update dependencies
- Prefix `DispatchQueue` to avoir collisions on iOS
- Bump major version as it contains a breaking change

## Motivation and Context
Make Trikot available for Kotlin `1.4.30`, making it compatible with JetPack Compose

## How Has This Been Tested?
The tests were run

## Types of changes
Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -8,3 +8,4 @@ org.gradle.jvmargs=-Xmx3072m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryErro
 org.gradle.caching=true
 kotlin.js.compiler=ir
 kotlin.js.ir.dce=true
+kotlin.mpp.stability.nowarn=true
```
Comment (marcantoinefortier) [inline gradle.properties:11]: 👍

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+buildscript {
+    repositories {
+        google()
+        maven(url = "https://plugins.gradle.org/m2/")
+    }
+
+    dependencies {
+        this.classpath("com.android.tools.build:gradle:4.1.2")
```
Comment (Boubalou) [inline build.gradle.kts:8]: this?

**Review by martingagnon — APPROVED** [review]

File: build.gradle.kts (hunk 2 above)
Comment (antoinelamy) [reply build.gradle.kts:8]: That may just be a leftover from my auto-complete attempt. Auto-completion in Gradle script 😍

Comment (antoinelamy) [reply build.gradle.kts]: Fixed ✅

**Pull request merged**